scipy/sparse/base.py

```diff
@@ -767,48 +845,129 @@ def sum(self, axis=None):
 
         if axis is None:
             # sum over rows and columns
-            return (self * np.asmatrix(np.ones((n, 1), dtype=res_dtype))).sum()
+            return (self * np.asmatrix(np.ones(
+                (n, 1), dtype=res_dtype))).sum(
+                dtype=dtype, out=out)
+
+        axis_type = type(axis)
+
+        # In NumPy, you can pass in tuples for 'axis', so let's make
+        # it explicit that you cannot do that with sparse matrices
+        if axis_type == tuple:
```

Comment:
It might be better to just add support for axis tuples, and avoid the inconsistency. The implementation shouldn't be too complex. We could pull out the code that processes the axis argument into a helper function, because this might be useful in other places as well.